pkg/metrics/collector/pod_ip_metrics.go

```diff
@@ -185,14 +222,43 @@ func (c *podIpMetricsCollector) setupDirectoryWatcher(dir string) error {
 
 		for {
 			select {
-			case _, ok := <-watcher.Events:
+			case e, ok := <-watcher.Events:
 				if !ok {
 					glog.Error("watcher is not ok")
 					return
 				}
 				if err := c.listIpAddresses(dir); err != nil {
 					return
 				}
+				names := strings.Split(e.Name, "/")
+				f := names[len(names)-1]
+				reusedIp, ok := reusedMap[f]
+				if e.Op&fsnotify.Remove == fsnotify.Remove {
+					if !ok {
+						reusedMap[f] = &ipReused{time.Now(), 0}
+					} else {
+						reusedIp.podKilledTimestamp = time.Now()
+					}
+				} else if e.Op&fsnotify.Create == fsnotify.Create {
+					if ok {
```

Comment:
Do you mean we need to clean up the map after every 60 seconds threshold? 
For this block the map only gets updated when the ip is reused.

Reply:
Actually, disregard this comment. We can remove the entries from the map for all IPs that are currently in use (which is what I was trying to say). But it does not save us much. However, we do want to make sure that we are not adding IPv6 entries to the map. Otherwise we will end up growing the map to billions of entries (theoretically).

Reply:
Understand, I keep the original logic.